### PR TITLE
[JBPM-10016] Drools/jBPM integration: high number of instances waitin…

### PR DESCRIPTION
**Pull request by mariofusco**
…g for signal adversely impacts execution time

### CODE REVIEW

**Review by elguardian — APPROVED** [review]

**Comment by gmunozfe** [comment]
@mariofusco, add the reference to the drools PR in the description: https://github.com/kiegroup/drools/pull/4196

**Review by gmunozfe — CHANGES_REQUESTED** [review]
It's not working as expected: the 3 threads are taking the same batch, instead of distributing the load among them, @elguardian could you take a look?

File: jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPASignalManager.java [hunk 1]
```diff
@@ -16,12 +16,14 @@
 
 package org.jbpm.persistence.processinstance;
```
Comment (elguardian) [inline jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPASignalManager.java:17]: @mariofusco  do you mind to remove this file change. I will treat the batch system in a different PR.

Comment (elguardian) [reply jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPASignalManager.java:17]: yes. it should

Comment (elguardian) [reply jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPASignalManager.java:17]: @mariofusco

**Pull request merged**